src/controller/python/chip/internal/commisioner.py

```diff
@@ -0,0 +1,144 @@
+#
+#    Copyright (c) 2021 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+from chip.configuration import GetLocalNodeId
+from chip.native import NativeLibraryHandleMethodArguments, GetLibraryHandle
+from ctypes import c_uint64, c_uint32, c_uint16, c_char_p
+from typing import Optional
+from chip.internal.types import NetworkCredentialsRequested, OperationalCredentialsRequested, PairingComplete
+import ctypes
+
+# Not using c_void_p directly is IMPORTANT. Python auto-casts c_void_p
+# to intergers and this can cause 32/64 bit issues.
```

Comment:
typo: integers

Also WTF ctypes :(

Good find.

Reply:
This was painful to debug.